src/writers.ts

```diff
@@ -1,22 +1,39 @@
 import { evm } from '@snapshot-labs/checkpoint';
 import { Payment } from '../.checkpoint/models';
+import { getJSON } from './utils';
+import tokens from './payment_tokens.json';
+
+function getTokenSymbol(tokenAddress: string, chain: string) {
+  return tokens[chain][tokenAddress];
+}
 
 export function createEvmWriters(indexerName: string) {
   const handlePaymentReceived: evm.Writer = async ({ block, tx, event }) => {
     console.log('In handlePaymentReceived');
     if (!block || !event) return;
 
     const sender = event.args.sender;
-    const token = event.args.token;
-    const amount = event.args.amount;
+    const tokenAddress = event.args.token.toLowerCase();
+    const amountRaw = BigInt(event.args.amount);
+    const amountDecimal = amountRaw / BigInt(1e6); // Transforms 1990000 to 1.99. We used 1e6 because USDC and USDT have 6 decimals.
```

Comment:
This doesn't work because both operands are BigInts then BigInt division is used which cannot be decimals (BigInts are integers).

```js
BigInt(1990000) / BigInt(1e6)
// equals to 1n
```

Reply:
e42ea825a7afea3bd83e0b80ddffd045a9ae79e3

I used `number` casting because from what I understood, it won't be lossy until 2^53, which should be enough?

The number is transformed to its decimals form but not truncated (i.e 1990000 will be "1.990000" and not "1.99"), so I removed the comment.